subsys/net/lib/sockets/socketpair.c

```diff
@@ -56,6 +56,11 @@ __net_socket struct spair {
 	uint8_t buf[CONFIG_NET_SOCKETPAIR_BUFFER_SIZE];
 };
 
+#ifdef CONFIG_NET_SOCKETPAIR_STATIC
+static struct spair storage[CONFIG_NET_SOCKETPAIR_MAX * 2];
```

Comment:
Would it make sense to use `k_mem_slab` to hide the allocation details? It could simplify the allocation/deallocation of static objects a bit.

Reply:
sys_bitarray is also useful for this sort of thing.

I just wish the lock for the bitarray was external to the structure.